src/main/java/com/direwolf20/buildinggadgets/common/building/placement/IPlacementSequence.java

```diff
@@ -0,0 +1,71 @@
+package com.direwolf20.buildinggadgets.common.building.placement;
+
+import com.direwolf20.buildinggadgets.common.building.Region;
+import com.google.common.collect.ImmutableList;
+import net.minecraft.util.math.BlockPos;
+
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Represents all block that should be changed for the build.
+ * <p>
+ * The yielding position should be inside the {@link #getBoundingBox()}. They do not have to be continuous, or yielding
+ * in any order, however it should not yield any repeating positions and have a finite number of possible results.
+ * </p>
+ */
+public interface IPlacementSequence extends Iterable<BlockPos> {
+
+    /**
+     * The bounding box such that all yielding positions are inside the region.
+     */
+    Region getBoundingBox();
+
+    /**
+     * @return {@code false} if this PlacementSequence definitely doesn't contain the specified Position. Otherwise {@code true}.
+     * @implSpec If the computation is costly, then this {@code IPlacementSequence} may choose to return always {@code true}
+     */
+    boolean contains(int x, int y, int z);
```

Comment:
please call it mayContain and actually return true, when the computation is costly or difficult (aka Grid, which violates the contract of this Method)